src/Accordion.tsx

```diff
@@ -36,25 +54,23 @@ const Accordion = React.forwardRef((props, ref) => {
     activeKey: 'onSelect',
   });
 
-  bsPrefix = useBootstrapPrefix(bsPrefix, 'accordion');
-
   return (
-    <AccordionContext.Provider value={activeKey}>
-      <SelectableContext.Provider value={onSelect}>
+    <AccordionContext.Provider value={activeKey || null}>
+      <SelectableContext.Provider value={onSelect || null}>
         <Component
           ref={ref}
           {...controlledProps}
-          className={classNames(className, bsPrefix)}
+          className={classNames(className, useBootstrapPrefix(bsPrefix, 'accordion'))}
```

Comment:
Should we be calling hooks inside the return statement of the component? I'm not sure if this is safe to do.